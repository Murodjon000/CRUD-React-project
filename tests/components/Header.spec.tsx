import shallowSetup from "./mock.spec";
import toJson from "enzyme-to-json";

test("should render Header component", () => {
  const { headerWrapper } = shallowSetup();

  expect(headerWrapper.find("header").hasClass("header-test")).toBe(true);
  expect(headerWrapper.find(".header__box_innerBox-logo").text()).toBe(
    "ProjectList"
  );
  expect(headerWrapper.find(".header-btn").text()).toBe("Sign up");
});

test("should render Header component Snapshot", () => {
  const { headerWrapper } = shallowSetup();

  expect(toJson(headerWrapper)).toMatchSnapshot();
});
