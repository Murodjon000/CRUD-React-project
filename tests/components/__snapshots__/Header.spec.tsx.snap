// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Details component Snapshot 1`] = `
<div
  className="project__details-wrapper"
>
  <div
    className="project__details-wrapper-image"
  >
    <img
      alt=""
      className="test-img"
      src="https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg"
    />
  </div>
  <div
    className="project__details-wrapper-box"
  >
    <h1
      className="test-title"
    >
      Where can I get some?
    </h1>
    <p
      className="test-desc"
    >
       
      ðŸ‘€ PageMaker including versions of Lorem Ipsum.
    </p>
    <button
      className="test-btn"
      onClick={[Function]}
      type="button"
    >
      Back to home
    </button>
  </div>
</div>
`;

exports[`should render Header component Snapshot 1`] = `
<header
  className="header-test"
>
  <div
    className="header__box"
  >
    <div
      className="header__box_innerBox"
    >
      <Link
        className="header__box_innerBox-logo"
        to="/"
      >
        ProjectList
      </Link>
    </div>
    <div>
      <button
        className="header-btn"
        onClick={[Function]}
      >
        Sign up
      </button>
      <span
        className="menuBars__wrapper"
        onClick={[Function]}
      >
        <i
          className="fa fa-bars"
        />
      </span>
    </div>
  </div>
  <div />
</header>
`;
